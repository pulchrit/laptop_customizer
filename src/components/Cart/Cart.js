import React from 'react';
import "../../App.css";
import CartItem from "../CartItem/CartItem";
import Total from "../Total/Total";

class Cart extends React.Component {
    
    render() {

        const summary = Object.keys(this.props.selectedState)
            .map(key => 
                <CartItem 
                    key={key}
                    selectedFeatureName={key}
                    selectedState={this.props.selectedState}
                />
            );
            
        return (
            <section className="main__summary">
                <h3>NEW GREENLEAF 2018</h3>
                {summary}
                <Total selectedState={this.props.selectedState} />
            </section>
        );
    }
}

export default Cart;

