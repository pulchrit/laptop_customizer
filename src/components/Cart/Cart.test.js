import React from 'react';
import ReactDOM from 'react-dom';
import Cart from "./Cart";
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';


describe("Cart component", () => {
    
    const selected = {
        Processor: {
            name: '17th Generation Intel Core HB (7 Core with donut spare)',
            cost: 700
        },
        "Operating System": {
            name: 'Ubuntu Linux 16.04',
            cost: 200
        },
        "Video Card": {
            name: 'Toyota Corolla 1.5v',
            cost: 1150.98
        },
        Display: {
            name: '15.6" UHD (3840 x 2160) 60Hz Bright Lights and Knobs',
            cost: 1500
        }
    };
    

    // smoke test
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Cart />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    // snapshot test
    it('renders cart with selected items', () => {
        const wrapper = shallow(<Cart selectedState={selected} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    // snapshot test?? Would this test make sense to add with a blank state object? 
    // I can't tell if there would be a default case where no selections were made...
    /* it('renders cart with no items', () => {

    }); */

});
