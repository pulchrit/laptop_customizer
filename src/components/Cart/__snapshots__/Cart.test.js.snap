// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart component renders cart with selected items 1`] = `
<section
  className="main__summary"
>
  <h3>
    NEW GREENLEAF 2018
  </h3>
  <CartItem
    key="Processor"
    selectedFeatureName="Processor"
    selectedState={
      Object {
        "Display": Object {
          "cost": 1500,
          "name": "15.6\\" UHD (3840 x 2160) 60Hz Bright Lights and Knobs",
        },
        "Operating System": Object {
          "cost": 200,
          "name": "Ubuntu Linux 16.04",
        },
        "Processor": Object {
          "cost": 700,
          "name": "17th Generation Intel Core HB (7 Core with donut spare)",
        },
        "Video Card": Object {
          "cost": 1150.98,
          "name": "Toyota Corolla 1.5v",
        },
      }
    }
  />
  <CartItem
    key="Operating System"
    selectedFeatureName="Operating System"
    selectedState={
      Object {
        "Display": Object {
          "cost": 1500,
          "name": "15.6\\" UHD (3840 x 2160) 60Hz Bright Lights and Knobs",
        },
        "Operating System": Object {
          "cost": 200,
          "name": "Ubuntu Linux 16.04",
        },
        "Processor": Object {
          "cost": 700,
          "name": "17th Generation Intel Core HB (7 Core with donut spare)",
        },
        "Video Card": Object {
          "cost": 1150.98,
          "name": "Toyota Corolla 1.5v",
        },
      }
    }
  />
  <CartItem
    key="Video Card"
    selectedFeatureName="Video Card"
    selectedState={
      Object {
        "Display": Object {
          "cost": 1500,
          "name": "15.6\\" UHD (3840 x 2160) 60Hz Bright Lights and Knobs",
        },
        "Operating System": Object {
          "cost": 200,
          "name": "Ubuntu Linux 16.04",
        },
        "Processor": Object {
          "cost": 700,
          "name": "17th Generation Intel Core HB (7 Core with donut spare)",
        },
        "Video Card": Object {
          "cost": 1150.98,
          "name": "Toyota Corolla 1.5v",
        },
      }
    }
  />
  <CartItem
    key="Display"
    selectedFeatureName="Display"
    selectedState={
      Object {
        "Display": Object {
          "cost": 1500,
          "name": "15.6\\" UHD (3840 x 2160) 60Hz Bright Lights and Knobs",
        },
        "Operating System": Object {
          "cost": 200,
          "name": "Ubuntu Linux 16.04",
        },
        "Processor": Object {
          "cost": 700,
          "name": "17th Generation Intel Core HB (7 Core with donut spare)",
        },
        "Video Card": Object {
          "cost": 1150.98,
          "name": "Toyota Corolla 1.5v",
        },
      }
    }
  />
  <Total
    selectedState={
      Object {
        "Display": Object {
          "cost": 1500,
          "name": "15.6\\" UHD (3840 x 2160) 60Hz Bright Lights and Knobs",
        },
        "Operating System": Object {
          "cost": 200,
          "name": "Ubuntu Linux 16.04",
        },
        "Processor": Object {
          "cost": 700,
          "name": "17th Generation Intel Core HB (7 Core with donut spare)",
        },
        "Video Card": Object {
          "cost": 1150.98,
          "name": "Toyota Corolla 1.5v",
        },
      }
    }
  />
</section>
`;
